adapters/insticator/insticator.go

```diff
@@ -0,0 +1,526 @@
+	"encoding/json"
+	"fmt"
+	"log"
+	"math"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+// type adapter struct {
+// 	endpoint string
+// }
+
+type Ext struct {
+	Insticator impInsticatorExt `json:"insticator"`
+}
+
+type insticatorUserExt struct {
+	Eids    []openrtb2.EID  `json:"eids,omitempty"`
+	Data    json.RawMessage `json:"data,omitempty"`
+	Consent string          `json:"consent,omitempty"`
+}
+
+type impInsticatorExt struct {
+	AdUnitId    string `json:"adUnitId,omitempty"`
+	PublisherId string `json:"publisherId,omitempty"`
+}
+
+type adapter struct {
+	endpoint string
+}
+
+type reqExt struct {
+	Insticator *reqInsticatorExt `json:"insticator,omitempty"`
+}
+
+type reqInsticatorExt struct {
+	Caller []InsticatorCaller `json:"caller,omitempty"`
+}
+
+type InsticatorCaller struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+// CALLER Info used to track Prebid Server
+// as one of the hops in the request to exchange
+var CALLER = InsticatorCaller{"Prebid-Server", "n/a"}
+
+type bidExt struct {
+	Insticator bidInsticatorExt `json:"insticator,omitempty"`
+}
+
+type bidInsticatorExt struct {
+	MediaType string `json:"mediaType,omitempty"`
+}
+
+// Placeholder for the actual openrtb2.Video struct
+type Video struct {
+	W              *int     `json:"w"`
+	H              *int     `json:"h"`
+	MIMEs          []string `json:"mimes"`
+	Placement      int      `json:"placement"`
+	Plcmt          int      `json:"plcmt"`
+	MinDuration    *int     `json:"minduration"`
+	MaxDuration    *int     `json:"maxduration"`
+	Protocols      []int    `json:"protocols"`
+	StartDelay     *int     `json:"startdelay"`
+	Linearity      *int     `json:"linearity"`
+	Skip           *int     `json:"skip"`
+	SkipMin        *int     `json:"skipmin"`
+	SkipAfter      *int     `json:"skipafter"`
+	Sequence       *int     `json:"sequence"`
+	Battr          []int    `json:"battr"`
+	MaxExtended    *int     `json:"maxextended"`
+	MinBitrate     *int     `json:"minbitrate"`
+	MaxBitrate     *int     `json:"maxbitrate"`
+	PlaybackMethod []int    `json:"playbackmethod"`
+	PlaybackEnd    *int     `json:"playbackend"`
+	Delivery       []int    `json:"delivery"`
+	Pos            *int     `json:"pos"`
+	API            []int    `json:"api"`
+}
+
+type BadInput struct {
+	Message string
+}
+
+func (e *BadInput) Error() string {
+	return e.Message
+}
+
+// Validation functions
+func isInteger(value interface{}) bool {
+	switch v := value.(type) {
+	case int:
+		return true
+	case float64:
+		return v == float64(int(v))
+	case string:
+		_, err := strconv.Atoi(v)
+		return err == nil
+	default:
+		return false
+	}
+}
+
+func isArrayOfNums(value interface{}) bool {
+	switch v := value.(type) {
```

Comment:
Based on https://github.com/prebid/prebid-server/pull/3806/files#r1705364786 comment this function is not required.